core/src/main/java/com/rudderstack/android/sdk/core/FlushUtils.java

```diff
@@ -0,0 +1,244 @@
+                return false;
+            }
+        }
+        return true;
+
+
+    }
+
+    /*
+     * flush events payload to server and return response as String
+     * */
+    static Utils.NetworkResponses flushEventsToServer(String payload, String dataPlaneUrl, String authHeaderString, String anonymousIdHeaderString) {
+        try {
+            if (TextUtils.isEmpty(authHeaderString)) {
+                RudderLogger.logError("EventRepository: flushEventsToServer: WriteKey was not correct. Aborting flush to server");
+                return null;
+            }
+
+            // get endPointUrl form config object
+            String dataPlaneEndPoint = dataPlaneUrl + "v1/batch";
+            RudderLogger.logDebug("EventRepository: flushEventsToServer: dataPlaneEndPoint: " + dataPlaneEndPoint);
+
+            // create url object
+            URL url = new URL(dataPlaneEndPoint);
+            // get connection object
+            HttpURLConnection httpConnection = (HttpURLConnection) url.openConnection();
+            // set connection object to return output
+            httpConnection.setDoOutput(true);
+            //  set content type for network request
+            httpConnection.setRequestProperty("Content-Type", "application/json");
+            // set authorization header
+            httpConnection.setRequestProperty("Authorization", String.format(Locale.US, "Basic %s", authHeaderString));
+            // set anonymousId header for definitive routing
+            httpConnection.setRequestProperty("AnonymousId", anonymousIdHeaderString);
+            // set request method
+            httpConnection.setRequestMethod("POST");
+            // get output stream and write payload content
+            OutputStream os = httpConnection.getOutputStream();
+            synchronized (FLUSH_LOCK) {
+                OutputStreamWriter osw = new OutputStreamWriter(os, "UTF-8");
+                osw.write(payload);
+                osw.flush();
+                osw.close();
+                os.close();
+                // create connection
+                httpConnection.connect();
+            }
+            // get input stream from connection to get output from the server
+            if (httpConnection.getResponseCode() == 200) {
+                BufferedInputStream bis = new BufferedInputStream(httpConnection.getInputStream());
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                int res = bis.read();
+                // read response from the server
+                while (res != -1) {
+                    baos.write((byte) res);
+                    res = bis.read();
+                }
+                // finally return response when reading from server is completed
+                if (baos.toString().equalsIgnoreCase("OK")) {
```

Comment:
Let's get away from checking the response string. 
200 response code should be sufficient.

Reply:
Ok, we can do that.